Alignment/OfflineValidation/plugins/OverlapValidation.cc

```diff
@@ -0,0 +1,876 @@
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "CommonTools/Statistics/interface/ChiSquaredProbability.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "CalibTracker/SiStripCommon/interface/SiStripDetInfoFileReader.h"
+
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetType.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/TrackerGeometryBuilder/interface/RectangularPixelTopology.h"
+
+#include  "DataFormats/TrackReco/interface/TrackFwd.h"
+#include  "DataFormats/TrackReco/interface/Track.h"
+#include  "DataFormats/TrackReco/interface/TrackExtra.h"
+#include  "DataFormats/TrackReco/interface/TrackExtraFwd.h"
+#include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "DataFormats/DetId/interface/DetId.h" 
+//#include "DataFormats/SiStripDetId/interface/TIBDetId.h"
+//#include "DataFormats/SiStripDetId/interface/TOBDetId.h"
+//#include "DataFormats/SiStripDetId/interface/TIDDetId.h"
+//#include "DataFormats/SiStripDetId/interface/TECDetId.h"
+#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
+#include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonDetUnit/interface/GluedGeomDet.h"
+#include "TrackingTools/PatternTools/interface/Trajectory.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHit.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/TSiStripMatchedRecHit.h"
+#include "RecoTracker/TransientTrackingRecHit/interface/ProjectedRecHit2D.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "SimTracker/TrackerHitAssociation/interface/TrackerHitAssociator.h"
+
+#include "TrackingTools/GeomPropagators/interface/AnalyticalPropagator.h"
+#include "TrackingTools/AnalyticalJacobians/interface/JacobianLocalToCurvilinear.h"
+#include "TrackingTools/AnalyticalJacobians/interface/JacobianCurvilinearToLocal.h"
+#include "TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h"
+#include "TrackingTools/TrackFitters/interface/TrajectoryStateCombiner.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
+#include "Alignment/TrackerAlignment/interface/TrackerAlignableId.h"
+#include "TFile.h"
+#include "TTree.h"
+
+#include <vector>
+#include <utility>
+
+using namespace std;
+//
+// class decleration
+//
+
+
+class OverlapValidation : public edm::EDAnalyzer {
```

Comment:
please avoid using a legacy module.
if this cannot be made multi-thread friendly make it at least a `one::EDAnalyzer`:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes

Reply:
I went with `one` for now.  As part of the condor migration maybe someone can make all the validation tools thread safe but it's too much for now.